Modules/Cart/Http/Services/CartService.php

```diff
@@ -119,56 +123,62 @@ public function add(array $request): array
                         $styleArr[] = $variant->value2;
                         $styleArr[] = $variant->value1;
                     }
-                    $tprice = 0;
+                    $tPrice = 0;
                     foreach ($qtyArr as $qk => $qv) {
-                        $tprice += $priceArr[$qk] * $qv;
+                        $tPrice += $priceArr[$qk] * $qv;
                     }
-                    $alreadyCart->price = $tprice;
+                    $alreadyCart->price = $tPrice;
                     $alreadyCart->style_name = rtrim(implode(',', $styleArr), ',');
                     $alreadyCart->style_group_name = rtrim(implode(',', $styleGrpArr), ',');
                     $alreadyCart->reference = serialize($optionsArr);
                 }
                 if ($productType == 'PREPACK') {
-                    $prepackVariant = ProductPrepack::where('id', $request['prepack_id'])->first();
-                    $alreadyCart->price = $prepackVariant->packs_price;
-                    $alreadyCart->style_name = $prepackVariant->style;
-                    $alreadyCart->style_group_name = $prepackVariant->size_ratio . ';' . $prepackVariant->size_range;
+                    $pPackVariant = ProductPrepack::where('id', $preID)->first();
+                    $alreadyCart->price = $pPackVariant->packs_price;
+                    $alreadyCart->style_name = $pPackVariant->style;
+                    $alreadyCart->style_group_name = $pPackVariant->size_ratio . ';' . $pPackVariant->size_range;
+                    if ($variant->stock < $alreadyCart->quantity || $variant->stock <= 0) {
+
+                        return ['res' => false, 'msg' => 'Stock not sufficient!.', 'data' => ""];
+
+                    }
                 }
                 if ($productType == 'SINGLE_PRODUCT') {
                     if ($variant->stock < $alreadyCart->quantity || $variant->stock <= 0) {
-                        $response = ['res' => false, 'msg' => 'Stock not sufficient!.', 'data' => ""];
-                        return ($response);
+
+                        return ['res' => false, 'msg' => 'Stock not sufficient!.', 'data' => ""];
+
                     }
                 }
             } else {
                 if ($product->stock < $alreadyCart->quantity || $product->stock <= 0) {
-                    $response = ['res' => false, 'msg' => 'Stock not sufficient!.', 'data' => ""];
-                    return ($response);
+
+                    return ['res' => false, 'msg' => 'Stock not sufficient!.', 'data' => ""];
+
                 }
             }
             $alreadyCart->save();
         } else {
             $cart = new Cart;
 
-            $cart->user_id = $request['user_id'];
+            $cart->user_id = $request->user_id;
```

Comment:
$cart->user_id = $request->user_id; should be changed to
$cart->user_id = auth()->user()->id;

Reply:
Done

Reply:
Done